src/diffusers/schedulers/scheduling_euler_discrete.py

```diff
@@ -362,57 +392,113 @@ def set_timesteps(
             raise ValueError(
                 "Cannot set `timesteps` with `config.timestep_type = 'continuous'` and `config.prediction_type = 'v_prediction'`."
             )
+        if timesteps is not None and self.config.use_flow_match:
+            # TODO: `timesteps / self.config.num_train_timesteps` to get sigmas?
+            raise ValueError("Cannot set `timesteps` with `config.use_flow_match = True`.")
+
+        if self.config.use_dynamic_shifting and mu is None:
+            raise ValueError(" you have a pass a value for `mu` when `use_dynamic_shifting` is set to be `True`")
 
         if num_inference_steps is None:
             num_inference_steps = len(timesteps) if timesteps is not None else len(sigmas) - 1
         self.num_inference_steps = num_inference_steps
 
-        if sigmas is not None:
+        if self.config.final_sigmas_type == "sigma_min" and not self.config.use_flow_match:
+            sigma_last = ((1 - self.alphas_cumprod[0]) / self.alphas_cumprod[0]) ** 0.5
+        elif self.config.final_sigmas_type == "zero":
+            sigma_last = 0
+        elif self.config.invert_sigmas:
+            sigma_last = 1
+        else:
+            raise ValueError(
+                f"`final_sigmas_type` must be one of 'zero', or 'sigma_min', but got {self.config.final_sigmas_type}"
+            )
+
+        if sigmas is not None and not self.config.use_flow_match:
             log_sigmas = np.log(np.array(((1 - self.alphas_cumprod) / self.alphas_cumprod) ** 0.5))
             sigmas = np.array(sigmas).astype(np.float32)
             timesteps = np.array([self._sigma_to_t(sigma, log_sigmas) for sigma in sigmas[:-1]])
+        elif sigmas is not None and self.config.use_flow_match:
+            sigmas = np.array(sigmas).astype(np.float32)
+            if self.config.use_flow_match:
+                if self.config.use_dynamic_shifting:
+                    sigmas = self.time_shift(mu, 1.0, sigmas)
+                else:
+                    sigmas = self.config.shift * sigmas / (1 + (self.config.shift - 1) * sigmas)
 
-        else:
+            if self.config.use_karras_sigmas:
```

Comment:
we have this argument `sigmas` is to accept a custom `sigmas` schedule from user, i.e. they should either pass a custom `sigmas` or choose to use one of the pre-set sigma schedules (e.g. karras, beta, exponential) ; 

so we should not have this logic here
```python
if sigma is not None and self.config.use_flow_match:
    ...
    if self.config.use_karras_sigmas:
          ...
```

Reply:
Yes this was due to how pipelines are currently set up and I wanted to test the noise schedules without modifying the pipelines, some of the logic is the same as existing FlowMatchEuler which applies shifting etc to the supplied sigmas, karras etc might actually work better when applied before the scaling so the future refactor will be really useful here.